### PR TITLE
[DNM]update kvrpc

### PR DESCRIPTION
**Pull request by morefreeze**
fix pingcap/tikv#218
@BusyJay PTAL. Please give some advice about `KeyAddress`.

### CODE REVIEW

File: generate_go.sh [hunk 1]
```diff
@@ -15,8 +15,12 @@ for file in `ls *.proto`
...
 
 echo "generate go code..."
+ret=0
 for file in `ls *.proto`  
     do
     base_name=$(basename $file ".proto")
     protoc --go_out=$GO_OUT_M:../pkg/$base_name $file
-done
\ No newline at end of file
+    last_ret=$?
```
Comment (BusyJay) [inline generate_go.sh]: Or just use `|| exit $?`?

File: generate_rust.sh [hunk 2]
```diff
@@ -4,9 +4,13 @@ echo "generate rust code..."
 protoc --rust_out ../src *.proto
 
 echo "extern crate protobuf;" > ../src/lib.rs
+ret=0
```
Comment (BusyJay) [inline generate_rust.sh]: What if protoc failed?

File: generate_go.sh (hunk 1 above)
Comment (morefreeze) [inline generate_go.sh]: Good idea. I think `|| ret = $?` is enough. I still wanna protoc each file, but if some failed just exit no-zero.

Comment (siddontang) [inline generate_go.sh]: do we need to check this? if failed, protoc will show the error message to console.

File: proto/kvrpcpb.proto [hunk 3]
```diff
@@ -1,61 +1,157 @@
...
+    optional KeyAddress key_address = 1;
+    optional uint64 start_version = 2;
 }
+
+message CmdCleanupResponse {
+    enum ResultType {
+        Ok = 1;
+        Retryable = 2;
+        Committed = 3;
+        Rollbacked = 4;
```
Comment (siddontang) [inline proto/kvrpcpb.proto]: RolledBack

**Comment by morefreeze** [comment]
@disksing PTAL

File: proto/kvrpcpb.proto [hunk 4]
```diff
@@ -1,61 +1,160 @@
...
+
 message CmdPrewriteRequest {
     repeated KvPair puts = 1;
-    repeated bytes dels = 2;
-    repeated bytes locks = 3;
-    required uint64 start_version = 4;
+    repeated KeyAddress dels = 2;
+    repeated KeyAddress locks = 3;
+    // primary_lock_key_address
+    optional KeyAddress primary_address = 4;
```
Comment (disksing) [inline proto/kvrpcpb.proto]: Server will not use primary lock as a key to query raft engine. So `bytes` is enough.

File: proto/kvrpcpb.proto [hunk 5]
```diff
@@ -1,61 +1,160 @@
...
+    optional bool ok = 1;
 }
+
+message CmdCleanupRequest {
+    optional KeyAddress key_address = 1;
+    optional uint64 start_version = 2;
+}
+
+message CmdCleanupResponse {
+    enum ResultType {
```
Comment (disksing) [inline proto/kvrpcpb.proto]: Why not merging it with the `ResultType` outside?

**Comment by BusyJay** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**